tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
+                qml.expval(single_term_obs_list[i])
+                for i in group
+                if not isinstance(single_term_obs_list[i], qml.Identity)
+            ]
+            assert tapes[group_idx].shots.total_shots == 100
+            assert tapes[group_idx].operations == [qml.X(0)]
 
-        tapes, fn = split_non_commuting(tape, grouping_strategy="wires")
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_wires_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(fn([[0.1, 0.2], 0.3, 0.4, 0.5]), [0.01, 0.06, 0.06, 0.16, 0.25])
+        assert fn(results) == expected_result
 
     @pytest.mark.parametrize(
-        "make_H",
+        "H",
         [
-            lambda coeffs, obs_list: qml.Hamiltonian(coeffs, obs_list),
-            lambda coeffs, obs_list: qml.sum(*(qml.s_prod(c, o) for c, o in zip(coeffs, obs_list))),
+            qml.Hamiltonian([0.1, 0.2, 0.3, 0.4, 0.5, 2.5], single_term_obs_list),
+            qml.dot([0.1, 0.2, 0.3, 0.4, 0.5, 2.5], single_term_obs_list),
         ],
     )
-    def test_grouping_strategies_single_hamiltonian(self, make_H):
-        """Tests that a single Hamiltonian or Sum is split correctly"""
-
-        coeffs = [0.1, 0.2, 0.3, 0.4, 0.5]
-        obs_list = single_term_obs_list
-        H = make_H(coeffs, obs_list)  # Tests that constant offset is handled
-
-        expected_tapes_no_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o)], shots=100) for o in obs_list
-        ]
-
-        expected_tapes_qwc_grouping = [
-            qml.tape.QuantumScript([], [qml.expval(o) for o in group], shots=100)
-            for group in single_term_qwc_groups
-        ]
-
-        coeffs, obs_list = coeffs + [0.6], obs_list + [qml.I()]
-        H = make_H(coeffs, obs_list)  # Tests that constant offset is handled
-
-        tape = qml.tape.QuantumScript([], [qml.expval(H)], shots=100)
-
-        tapes, fn = split_non_commuting(tape, grouping_strategy=None)
-        for actual_tape, expected_tape in zip(tapes, expected_tapes_no_grouping):
-            qml.assert_equal(actual_tape, expected_tape)
-        assert qml.math.allclose(fn([0.1, 0.2, 0.3, 0.4, 0.5]), 1.15)
+    @pytest.mark.parametrize("grouping_strategy", ["qwc", "default"])
+    def test_single_hamiltonian_grouping(self, H, grouping_strategy):
+        """Tests that a single Hamiltonian is split correctly."""
+
+        initial_tape = qml.tape.QuantumScript([qml.X(0)], [qml.expval(H)], shots=100)
+        tapes, fn = split_non_commuting(initial_tape, grouping_strategy=grouping_strategy)
+        assert H.grouping_indices == ((0, 3, 5), (1, 2, 4))
+
+        for group_idx, group in enumerate(H.grouping_indices):
+            assert tapes[group_idx].measurements == [
+                qml.expval(single_term_obs_list[i])
+                for i in group
+                if not isinstance(single_term_obs_list[i], qml.Identity)
+            ]
```

Comment:
Same state here as above --- splitting this would make it easier to understand.

Reply:
Resolved without comment.

Reply:
I've split this into multiple lines. It appears that if you directly click the link to the code from the comment, it takes you to an old version of the code. You can see the updated lines if you manually navigate to its location.